source/opt.d

```diff
@@ -1,31 +1,71 @@
 import std.getopt;
+import std.variant;
+import std.file;
+import std.stdio;
+import core.stdc.stdlib;
 
-void getoptions(string[] args)
-in
+struct Options {
+
+    ulong countEpoch;
+    float mutation;
+    string input;
+    float probability;
+
+    this(string inp, float mutate, ulong  count, float prob)
+    {
+        this.input = inp;
+        this.mutation = mutate;
+        this.countEpoch = count;
+        this.probability = prob;
+    }
+}
+
+class NoInputParams : Exception
 {
-    assert(args.length > 1, "Inputed empty paramters");
-    assert(args.length < 5, "Opps too many argmunt inputed");
+    this(string msg, string file = __FILE__, size_t line = __LINE__) {
+        super(msg, file, line);
+    }
 }
-body
+
+Options getOptions(string[] args)
 {
-    string data;
-    int count_epoch;
-    int mutation_number;
-    float probability = 0.0;
+    string input;
+    ulong countEpoch;
+    float mutationNumber = 0.99f;
+    float probability;
     auto helpInformation = getopt(
-        args,
-        "epoch|e","Counting epoch, number(int).", &count_epoch,
-        "mutation|m","Mutation, how large mutation should be(int).", &mutation_number,
-        "probability|p","Number(float) representing probability.", &probability,
-        "input|i","Input file with data.", &data
-    );
-    import std.stdio;
-    data.writeln;
+            args,
+            "input|i","This is the image you MUST input", &input,
+            "epoch|e","Counting epoch, number (ulong).", &countEpoch,
+            "probability|p","Number (float) representing probability.", &probability,
+            "mutation|m","Mutation, how large mutation is, value MUST be between 0.0 and 1", &mutationNumber
```

Comment:
- Mezery za názvem optinonl a popisem
- Povinný argumenty
- Proč to necpeš rovnou do tý Options struktury?